hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -157,6 +157,11 @@
       .withDocumentation("When enabled, populates all meta fields. When disabled, no meta fields are populated "
           + "and incremental queries will not be functional. This is only meant to be used for append only/immutable data for batch processing");
 
+  public static final ConfigProperty<String> HOODIE_TABLE_KEY_GENERATOR_CLASS = ConfigProperty
+      .key("hoodie.datasource.write.keygenerator.class")
```

Comment:
do we really write this as `hoodie.datasource` prefix?  This is a table level property, not just for the datasource write. lets fix this?

Reply:
my intention here is to just use the same config someone uses with spark datasource. Rather than introducing new configs. But guess we don't follow that for partition path nor record keys. Will fix it.

Reply:
I will follow similar naming convention as record key and partition path
hoodie.table.keygenerator.class